test/evp_test.c

```diff
@@ -2058,31 +1957,26 @@ struct keypair_test_data {
     EVP_PKEY *pubk;
 };
 
-static int keypair_test_init(struct evp_test *t, const char *pair)
+static int keypair_test_init(evp_test *t, const char *pair)
 {
     int rv = 0;
     EVP_PKEY *pk = NULL, *pubk = NULL;
     char *pub, *priv = NULL;
     const char *err = "INTERNAL_ERROR";
     struct keypair_test_data *data;
 
-    priv = OPENSSL_strdup(pair);
-    if (priv == NULL)
-        return 0;
-    pub = strchr(priv, ':');
-    if ( pub == NULL ) {
-        fprintf(stderr, "Wrong syntax \"%s\"\n", pair);
+    if (!TEST_ptr(priv = OPENSSL_strdup(pair))
+            || !TEST_ptr(pub = strchr(priv, ':')))
```

Comment:
So from that output, you will know exactly which keypair from `test/evptests.txt` was wrong?  If so, your fu is much stronger than mine...